lib/sidekiq_alive/server/base.rb

```diff
@@ -4,19 +4,34 @@ module SidekiqAlive
   module Server
     module Base
       SHUTDOWN_SIGNAL = "TERM"
+      QUIET_SIGNAL = "USR1"
 
-      def shutdown!
-        SidekiqAlive.logger.info("Shutting down SidekiqAlive web server")
-        Process.kill(SHUTDOWN_SIGNAL, @server_pid) unless @server_pid.nil?
-        Process.wait(@server_pid) unless @server_pid.nil?
+      # set web server to quiet mode
+      def quiet!
+        logger.info("[SidekiqAlive] Setting web server to quiet mode")
+        Process.kill(QUIET_SIGNAL, @server_pid) unless @server_pid.nil?
```

Comment:
Do we have to send the `QUIET_SIGNAL` here. The result of this would be dependent of the implementation of each server.
What if we store in the server instance a `@quiet = true` and do exactly the same logic you already implemented?

Reply:
@arturictus 

Yes, we need `QUIET_SIGNAL` because the web server runs in a forked process so the only way to "interact" with that process is via signals. We can't really store anything in the main process that would also be seen by the forked web server process.

Also, it doesn't really depend on implementation of each server, or rather we have 2 server implementations, the basic web server and rack based web servers and they both implement how they handle the `QUIET_SIGNAL` which for the most part is to simply set a time when `quite` mode got triggered so we have a safeguard of not being stuck in `quiet` mode indefinitely if `sidekiq` process hangs for example.

The only risk here is that we use `USR1` signal. In theory, if a user flow uses same signal, we may end up triggering quiet mode unintentionally. This could probably be documented to avoid confusion.

Reply:
I see, so it makes sense to send the signal.
It could be nice to have documentation, but I do not see the obligation to do it.
Thanks for the explanation.
Go ahead and merge if you feel confident about this PR

Reply:
I think we should be good, I have been running this code in my app pretty much since I created the PR.